### PR TITLE
Improve openshift and s2i configuration merge

### LINKED ISSUE #20939: Unrecognized system properties when using quarkus-openshift
### Describe the bug

Compatibility layer between s2i and openshift container image extensions does not cover system properties.
To elaborate:

When using `quarkus-container-image-openshift` all configuration properties of `quarkus-container-image-s2i` are support for compatibility reasons (so that it can act as a drop in replacement). So, all `quarkus.s2i.xxx` properties work when added to `application.properties` but not when passed as system properties.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #20939

### CODE REVIEW

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java [hunk 1]
```diff
@@ -147,4 +147,8 @@ public static OpenshiftConfig mergeConfig(OpenshiftConfig openshiftConfig, S2iCo
 
         return result;
     }
+
+    private static boolean hasProperty(String name, Set<String> names) {
+        return names.contains(name) || System.getProperties().containsKey(name);
```
Comment (geoand) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: What's the rationale behind this?

Comment (iocanel) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: We want to check if a property is present in the config object or is passed via system properties.

Comment (geoand) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: Sure, but why do we need to check systsme properties? The Config object already contains a source for system properties

Comment (geoand) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: This should not be the case... When the `Config` is created, some sources are added by default, like the `SysPropConfigSource`. If that is not happening here, we need to first understand why.

Comment (iocanel) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: My guess is that this is due to the fact that the config phase is BUILD_TIME.

Comment (iocanel) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: Hmmm, my previous comment doesn't make sense, as we are talking about BUILD_TIME.

Comment (iocanel) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: Using something like this `mvnDebug clean install -Dquarkus.s2i-base-jvm-image=test` does not seem to work.

Comment (iocanel) [reply extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java:152]: Let me elaborate. Debuging the build with a command like the one above results in the config not containing the property, even though the system properties source is listed.

It seems that the root cause is the System property not being visible at all.
I will close the PR.

**Pull request closed without merging**